internal/pkg/agent/application/upgrade/upgrade.go

```diff
@@ -272,12 +272,15 @@ func (u *Upgrader) Upgrade(ctx context.Context, version string, sourceURI string
 	// while the `previous` install is the currently executing elastic-agent that is no longer reachable via the symlink.
 	// After the restart at the end of the function, everything lines up correctly.
 	current := agentInstall{
+		parsedVersion: parsedVersion,
 		version:       version,
 		hash:          unpackRes.Hash,
 		versionedHome: unpackRes.VersionedHome,
 	}
 
+	previousParsedVersion := getParsedVersionWithFallback(u.log, release.VersionWithSnapshot(), agtversion.NewParsedSemVer(8, 13, 0, "", ""))
```

Comment:
Needs a comment explaining why 8.13.0 is hard coded here.

Reply:
Also it's not clear why we need a fallback. Under what circumstances would parsing `release.VersionWithSnapshot()` fail?

Reply:
It may be excess of caution on my side but release.VersionWithSnapshot() is a string returned by reading the package.version (which can be edited and made it to contain corrupt data) or referencing the BEATS_VERSION as a fallback. 
I didn't want to deal with a parsing error for the current version string at this stage of the upgrade process so I added the fallback to 8.13.0 which is the last version where the new update marker was introduced.

I could make the upgrade rollback here at the very end if we cannot parse the current version but I tried to give priority to the upgrade which at this point we managed to unpack, rotate links and copy state to...

Reply:
can you make a comment out of it. i bet i won't remember in a month

Reply:
I don't really like that this is hard-coded. Will this need to be updated when we create a new release? Why not use `version.Agent`?

Reply:
This has now been removed in favor of using the agent package version we initialize at startup.